API/src/Modules/Reports/Momentum.Reports.Infrastructure/Repositories/ReportRepository.cs

```diff
@@ -0,0 +1,54 @@
+﻿using Marten;
+using Momentum.Framework.Infrastructure.Repositories;
+using Momentum.Reports.Core.Models;
+using Momentum.Reports.Core.Repositories;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+
+namespace Momentum.Reports.Infrastructure.Repositories
+{
+    public class ReportRepository : GenericTimeTrackedRepository<Report>, IReportRepository
+    {
+        private readonly IDocumentStore _store;
+
+        public ReportRepository(IDocumentStore store) : base(store)
+        {
+            _store = store;
+        }
+
+        public async Task<int> GetTodayReportCount(Guid submitterId)
+        {
+            using var session = _store.QuerySession();
+
+            var numOfReportsSubmittedToday = await session.Query<Report>().CountAsync(x => x.SubmitterId == submitterId && x.CreatedAt > DateTime.Today);
+
+            return numOfReportsSubmittedToday;
+        }
+
+        public async Task<int> CountAllReports(bool resolved)
+        {
+            using var session = _store.QuerySession();
+
+            var numberOfReports = await session.Query<Report>().CountAsync(x => x.Resolved == resolved);
+
+            return numberOfReports;
+        }
+
+        public async Task<IReadOnlyList<Report>> GetAllReports(int? limit, uint offset, bool resolved)
```

Comment:
Change `int? limit` to `int limit = 5` (You'll have to make this the last parameter as optional parameters have to be last)

Reply:
If the limit parameter is changed to non-nullable, then it will default to 0. Then this default of 5 isn't applied and no reports are returned. For now I left this the same unless you have more suggestions